### PR TITLE
Fiddle with rewrites so dataflow graph remains identical

### PR DESCRIPTION
**Pull request by mshields822**
R: @dhalperi 

Something like this we think.
Will test later today.
Any tips for running tap without hand patching?

### CODE REVIEW

**Comment by dhalperi** [comment]
This looks pretty much right to me. Hope the tests agree!

**Comment by mshields822** [comment]
This matches changes needed for internal tests to pass.
Subsumes https://github.com/GoogleCloudPlatform/DataflowJavaSDK/pull/279

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineRunner.java [hunk 1]
```diff
@@ -362,15 +363,11 @@ public static DataflowPipelineRunner fromOptions(PipelineOptions options) {
       builder.put(Window.Bound.class, AssignWindows.class);
       // In streaming mode must use either the custom Pubsub unbounded source/sink or
       // defer to Windmill's built-in implementation.
-      builder.put(PubsubIO.Read.Bound.PubsubBoundedReader.class, UnsupportedIO.class);
-      builder.put(PubsubIO.Write.Bound.PubsubBoundedWriter.class, UnsupportedIO.class);
-      if (options.getExperiments() == null
```
Comment (dhalperi) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineRunner.java]: Did you mean to disable this experiment?

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineRunner.java [hunk 2]
```diff
@@ -421,10 +418,19 @@ public static DataflowPipelineRunner fromOptions(PipelineOptions options) {
       OutputT outputT = (OutputT) PCollection.createPrimitiveOutputInternal(
           pc.getPipeline(),
           transform instanceof GroupByKey
-              ? ((GroupByKey<?, ?>) transform).updateWindowingStrategy(pc.getWindowingStrategy())
-              : pc.getWindowingStrategy(),
+          ? ((GroupByKey<?, ?>) transform).updateWindowingStrategy(pc.getWindowingStrategy())
+          : pc.getWindowingStrategy(),
```
Comment (dhalperi) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineRunner.java]: fix indent back?

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/util/MovingFunction.java [hunk 3]
```diff
@@ -121,7 +115,7 @@ public void add(long nowMsSinceEpoch, long value) {
   }
 
   /**
-   * Return the minimum/maximum/sum of all retained values within {@link #samplePeriodMs}
+   * Return the minimum/maximum/sum of all retained values within samplePeriodMs
    * of {@code nowMsSinceEpoch}.
```
Comment (dhalperi) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/util/MovingFunction.java]: `{@code` ?

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/util/MovingFunction.java [hunk 4]
```diff
@@ -28,11 +28,6 @@
  */
 public class MovingFunction {
   /**
```
Comment (dhalperi) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/util/MovingFunction.java]: did you need to fix the `{@link #samplePeriodMs}` above?

**Comment by dhalperi** [comment]
Minor fixes in followup PR.

**Pull request merged**